storage/src/main/java/tech/pegasys/teku/storage/server/ChainStorage.java

```diff
@@ -132,17 +155,60 @@ public SafeFuture<Optional<SignedBeaconBlock>> getEarliestAvailableBlock() {
 
   @Override
   public SafeFuture<Optional<SignedBeaconBlock>> getFinalizedBlockAtSlot(final UInt64 slot) {
-    return SafeFuture.of(() -> database.getFinalizedBlockAtSlot(slot));
+    return SafeFuture.of(() -> database.getFinalizedBlockAtSlot(slot))
+        .thenCompose(this::unblindBlock);
+  }
+
+  private SafeFuture<Optional<SignedBeaconBlock>> unblindBlock(
+      final Optional<SignedBeaconBlock> maybeBlock) {
+    if (maybeBlock.isEmpty()) {
+      return SafeFuture.completedFuture(maybeBlock);
+    }
+    return unblindedBlock(maybeBlock.get()).thenApply(Optional::of);
+  }
+
+  private SafeFuture<SignedBeaconBlock> unblindedBlock(final SignedBeaconBlock block) {
```

Comment:
nit: Maybe this should be called unblindBlock so it's performing the action of unblinding the block.